src/components/datePicker/DatePickerDropdown.tsx

```diff
@@ -77,7 +77,7 @@ export class DatePickerDropdown extends React.Component<IDatePickerDropdownProps
 
   private handleDocumentClick = (e: MouseEvent) => {
     let dropdown: HTMLDivElement = ReactDOM.findDOMNode<HTMLDivElement>(this.dropdown);
-    if (!dropdown.contains(e.target as Node)) {
+    if (!dropdown.contains(e.target as Node) && this.props.isOpened) {
```

Comment:
I don't know if we really care about performance but we should check for the for the isOpened before checking if the dropdown contains the target